### PR TITLE
refactor contentHandler.adjustPageContent, extract functions.js, styles.css from htmlscripts.txt

### PR DESCRIPTION
**Pull request by shiraeeshi**

### CODE REVIEW

**Comment by babluboy** [comment]
@shiraeeshi Many thanks for your interest and contribution to Bookworm. Can you please mention a couple of lines on the changes for the PR and if possible any screen shots. That will give me a better idea on what the changes impact - will merge after you provide the details

File: src/contentHandler.vala [hunk 1]
```diff
@@ -286,26 +306,7 @@ public class BookwormApp.contentHandler {
             }
         }
         //complete the onload javascript string
-        BookwormApp.Bookworm.onLoadJavaScript.append ("\"");
-
-        //add onload javascript and css to body tag
-        if (pageContent.str.index_of ("<BODY") != -1) {
```
Comment (shiraeeshi) [inline src/contentHandler.vala]: contentHandler replaces old pageContent with the new one.

File: src/constants.vala.in [hunk 2]
```diff
@@ -162,8 +162,10 @@ namespace BookwormApp.Constants {
 
     public const string DICTIONARY_SCRIPT_LOCATION = INSTALL_TASKS_DIR + "/com.github.babluboy.bookworm.dictionary.sh";
     public const string SEARCH_SCRIPT_LOCATION = INSTALL_TASKS_DIR + "/com.github.babluboy.bookworm.search.sh";
-    public const string BOOKWORM_HTML_SCRIPT = "com.github.babluboy.bookworm.htmlscripts.txt";
```
Comment (shiraeeshi) [inline src/constants.vala.in]: javascript functions and css styles are mixed in a single file.

File: src/window.vala [hunk 3]
```diff
@@ -169,6 +170,19 @@ public class BookwormApp.AppWindow {
         webkitSettings.set_auto_load_images (true);
         aWebView = new WebKit.WebView.with_settings (webkitSettings);
         aWebView.set_zoom_level (BookwormApp.Settings.get_instance ().zoom_level);
+        aWebView.load_changed.connect((loadEvent) => {
```
Comment (shiraeeshi) [inline src/window.vala]: `onLoadJavaScript` is invoked using a webview API.

File: src/contentHandler.vala [hunk 4]
```diff
@@ -388,6 +389,16 @@ public class BookwormApp.contentHandler {
...
     public static void performStartUpActions () {
         debug ("[START] [FUNCTION:performStartUpActions]");
+        //load javascript data from resource if it has not been loaded already
+        if (BookwormApp.Bookworm.bookwormScripts == null || BookwormApp.Bookworm.bookwormScripts.length < 1) {
+            uint8[] bookwormScriptsData;
+            GLib.File.new_for_uri (BookwormApp.Constants.HTML_SCRIPT_FUNCTIONS_RESOURCE_LOCATION)
+                .load_contents (null, out bookwormScriptsData, null);
+            BookwormApp.Bookworm.bookwormScripts = (string)bookwormScriptsData;
+            WebKit.UserScript userScript = new WebKit.UserScript(BookwormApp.Bookworm.bookwormScripts, WebKit.UserContentInjectedFrames.ALL_FRAMES, WebKit.UserScriptInjectionTime.END, null, null);
+            BookwormApp.AppWindow.aWebView.get_user_content_manager().add_script(userScript);
```
Comment (shiraeeshi) [inline src/contentHandler.vala]: javascript code extracted from htmlscripts.txt is added to page using a webview API.

File: src/contentHandler.vala [hunk 5]
```diff
@@ -142,70 +142,90 @@ public class BookwormApp.contentHandler {
...
 
         //Remove the empty title if it is present
         pageContent.assign (pageContentStr.replace ("<title/>",""));
 
+        constructOnLoadJavascript(aBook, mode);
+
+        string currentBookwormStyles = createPageStyles(aBook, pageContentStr);
+
+        BookwormApp.AppWindow.aWebView.get_user_content_manager().remove_all_style_sheets();
+        BookwormApp.AppWindow.aWebView.get_user_content_manager().add_style_sheet(
```
Comment (shiraeeshi) [inline src/contentHandler.vala]: css styles extracted from htmlscripts.txt are added to page using a webview API.

File: src/constants.vala.in [hunk 6]
```diff
@@ -162,8 +162,10 @@ namespace BookwormApp.Constants {
...
-    public const string HTML_SCRIPT_RESOURCE_LOCATION = "resource:///" + RESOURCE_PREFIX + BOOKWORM_HTML_SCRIPT;
+    public const string BOOKWORM_HTML_SCRIPT_FUNCTIONS = "com.github.babluboy.bookworm.htmlscripts.functions.js";
+    public const string BOOKWORM_HTML_SCRIPT_STYLES = "com.github.babluboy.bookworm.htmlscripts.styles.css";
+    public const string HTML_SCRIPT_FUNCTIONS_RESOURCE_LOCATION = "resource:///" + RESOURCE_PREFIX + BOOKWORM_HTML_SCRIPT_FUNCTIONS;
+    public const string HTML_SCRIPT_STYLES_RESOURCE_LOCATION = "resource:///" + RESOURCE_PREFIX + BOOKWORM_HTML_SCRIPT_STYLES;
```
Comment (shiraeeshi) [inline src/constants.vala.in]: functions.js and styles.css files were extracted from htmlscripts.txt

**Comment by shiraeeshi** [comment]
@babluboy Thanks for creating an opportunity to work on an open source book reader.

I plan to gradually add little refactorings and occasionally a new functionality.

Now I'm working on a shortcuts dialog.

**Comment by babluboy** [comment]
Thanks I have merged. From a maintenance point of view it’s a good idea to have js and css separated - but will it be improving performance and load time? Will check for the same

**Comment by babluboy** [comment]
I had not been able to pay attention to bookworm for a couple of months due to my day job. But will start work soon on the pagination function.

Yes, the shortcuts will be a good to have addition within the app. At the moment they are on the website (https://babluboy.github.io/bookworm/). The in-app shortcuts can be based on OS as per #295

**Pull request merged**